### PR TITLE
Parses navvis origin file and converts into Capture::Lamar csv file 

### PR DESCRIPTION
**Pull request by vjlux**
* More tests missing

### CODE REVIEW

**Comment by sarlinpe** [comment]
Do new NavVis devices have GPS? I've never seen this CRS field.

**Comment by vjlux** [comment]
Navvis not. But u can align navvis data to georefeenced coordinate systems
in ivion and u get this fiels

**Review by pablospe — APPROVED** [review]
LGTM. Nit: remove the whit spaces.

File: scantools/capture/records.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 from typing import Dict, List, Set, TypeVar, Union
 import numpy as np
 
+from .pose import Pose
```
Comment (pablospe) [inline scantools/capture/records.py:11]: Needed? It seems the only change.

File: scantools/capture/records.py [hunk 2]
```diff
@@ -276,3 +277,6 @@ class RecordsBluetooth(RecordsArray[RecordBluetooth]):
             records[timestamp, sensor_id] = <RecordBluetooth>
     """
     record_type = RecordBluetooth
+
```
Comment (pablospe) [inline scantools/capture/records.py:280]: trailing spaces

File: scantools/run_navvis_to_capture.py [hunk 3]
```diff
@@ -98,6 +98,15 @@ def run(input_path: Path, capture: Capture, tiles_format: str, session_id: Optio
     images = RecordsCamera()
     rigs = Rigs() if export_as_rig else None
 
+    # Read the NavVis origin.json file if present and convert to Capture format.
+    if nv.load_origin():
+        origin_qvec, origin_tvec, origin_crs = nv.get_origin()
```
Comment (pablospe) [inline scantools/run_navvis_to_capture.py:103]: origin_crs is not used, right?

File: scantools/scanners/navvis/navvis.py [hunk 4]
```diff
@@ -677,6 +683,54 @@ def read_wifi(self):
...
+        tvec = [0, 0, 0]        
+        if self.__origin_data:
+            orientation = self.__origin_data['orientation']
+            position = self.__origin_data['position']
+            qvec = [orientation['w'], orientation['x'], orientation['y'], orientation['z']]
+            tvec = [position['x'], position['y'], position['z']]
+        return crs, qvec, tvec
+    
+    def load_origin(self):
+        """Tries loading the NavVis origin from file.
```
Comment (pablospe) [inline scantools/scanners/navvis/navvis.py:699]: Maybe this comment should be moved to the definition of `parse_navvis_origin_file()`?

File: scantools/scanners/navvis/origin_parser.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                f"{orientation['w']},"
+                f"{orientation['x']},"
+                f"{orientation['y']},"
+                f"{orientation['z']},"
+                f"{position['x']},"
+                f"{position['y']},"
+                f"{position['z']}\n")
+    return csv_str
+        
+
```
Comment (pablospe) [inline scantools/scanners/navvis/origin_parser.py:65]: nit: trailing spaces

File: scantools/scanners/navvis/origin_parser.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        with file_path.open() as f:
+            origin = json.load(f)
+            if not is_navvis_origin_valid(origin):
+                print("Invalid origin.json file", json.dumps(origin, indent=4))
+            return origin
+    except Exception as e:
+        print("Warning Failed reading origin.json file.", e)
+    return {}
+
+def convert_navvis_origin_to_csv(navvis_origin : dict):
```
Comment (pablospe) [inline scantools/scanners/navvis/origin_parser.py:44]: where is this function used? I only see this in the test.

**Comment by pablospe** [comment]
Please also update the `CAPTURE.md` to include the origin output:
https://github.com/microsoft/lamar-benchmark/blob/main/CAPTURE.md

File: scantools/capture/namedposes.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+from pathlib import Path
+from .pose import Pose
+from ..utils.io import read_csv, write_csv
+
+
+class NamedPoses(dict):
```
Comment (sarlinpe) [inline scantools/capture/namedposes.py:6]: Isn't this redundant with `capture.proc.GlobalAlignment`?

Comment (vjlux) [reply scantools/capture/namedposes.py]: ok.. I can write here now. was looking at it with the wrong account. I changed it using GlobalAlignment.

**Comment by vjlux** [comment]
> Please also update the `CAPTURE.md` to include the origin output: https://github.com/microsoft/lamar-benchmark/blob/main/CAPTURE.md

I wonder if I should just go with proc/global_alignment but this would be more involved.

File: CAPTURE.md [hunk 8]
```diff
@@ -23,6 +23,7 @@ location1/                                  # a Capture directory
 │   │   ├── sensors.txt                     # list of all sensors with specs
 │   │   ├── trajectories.txt                # pose for each (timestamp, sensor)
 │   │   ├── wifi.txt                        # list of wifi measurements
+|   |   ├── origins.txt                     # list of NavVis Session origins (name, alginment pose)
```
Comment (ahojnnes) [inline CAPTURE.md:26]:
```suggestion
|   |   ├── origins.txt                     # list of NavVis session origins (name, alignment pose) 
```

File: scantools/run_navvis_to_capture.py [hunk 9]
```diff
@@ -96,7 +96,7 @@ def run(input_path: Path, capture: Capture, tiles_format: str, session_id: Optio
     sensors = Sensors()
     trajectory = Trajectories()
     images = RecordsCamera()
-    rigs = Rigs() if export_as_rig else None
+    rigs = Rigs() if export_as_rig else None
```
Comment (ahojnnes) [inline scantools/run_navvis_to_capture.py:99]: unnecessary trailing space

File: scantools/run_navvis_to_capture.py [hunk 10]
```diff
@@ -231,12 +231,34 @@ def run(input_path: Path, capture: Capture, tiles_format: str, session_id: Optio
...
-        images=images, pointclouds=pointclouds, wifi=wifi_signals, bt=bluetooth_signals)
+        images=images, pointclouds=pointclouds, wifi=wifi_signals, bt=bluetooth_signals, 
+        origins=navvis_origin)
     capture.sessions[session_id] = session
     capture.save(capture.path, session_ids=[session_id])
 
+    # Read the NavVis origin.json file if present and use proc.GlobalAlignment to save it.
+    if nv.load_origin():
+        global_alginment_path = capture.data_path(session_id) / "origin.txt"
+        origin_qvec, origin_tvec, origin_crs = nv.get_origin()
```
Comment (ahojnnes) [inline scantools/run_navvis_to_capture.py:255]: origin_crs is intentionally unused?

File: scantools/run_navvis_to_capture.py [hunk 11]
```diff
@@ -231,12 +231,34 @@ def run(input_path: Path, capture: Capture, tiles_format: str, session_id: Optio
...
     session = Session(
         sensors=sensors, rigs=rigs, trajectories=trajectory,
-        images=images, pointclouds=pointclouds, wifi=wifi_signals, bt=bluetooth_signals)
+        images=images, pointclouds=pointclouds, wifi=wifi_signals, bt=bluetooth_signals, 
+        origins=navvis_origin)
     capture.sessions[session_id] = session
     capture.save(capture.path, session_ids=[session_id])
 
+    # Read the NavVis origin.json file if present and use proc.GlobalAlignment to save it.
+    if nv.load_origin():
```
Comment (ahojnnes) [inline scantools/run_navvis_to_capture.py:253]: Unclear to me why we load the origin twice here and above? A comment would be helpful to explain.

**Review by ahojnnes — CHANGES_REQUESTED** [review]

File: scantools/run_navvis_to_capture.py (hunk 11 above)
Comment (vjlux) [reply scantools/run_navvis_to_capture.py:253]: Changes not pushed. my bad

File: scantools/run_navvis_to_capture.py (hunk 10 above)
Comment (vjlux) [reply scantools/run_navvis_to_capture.py:255]: changes not pushed. my bad

**Review by pablospe — APPROVED** [review]

**Review by pablospe — APPROVED** [review]
lgtm

**Pull request merged**